registry/src/pages/upload.js

```diff
@@ -1,11 +1,18 @@
 import React, { useState } from "react";
 import { useCookies } from "react-cookie";
 import { Navigate } from "react-router-dom";
+import { useDispatch, useSelector } from "react-redux";
+import { uploadPackage } from "../store/actions/uploadActions";
+import Container from "react-bootstrap/Container";
+import Spinner from "react-bootstrap/Spinner";
 import "./upload.css";
 
 const PackageForm = () => {
   const [cookies, setCookie] = useCookies(["uuid"]);
-  const responseDiv = document.getElementById("error");
+  const dispatch = useDispatch();
+  const isLoading = useSelector((state) => state.upload.isLoading);
+  const message = useSelector((state) => state.upload.message);
+  document.getElementById("error").innerHTML = message;
```

Comment:
We should never do dom manipulation. Instead do conditional rendering using states like we are doing on login/signup page.